### PR TITLE
Configurable Maximum Data Points for Cartesian

### PR DESCRIPTION
**Pull request by skjindal93**

### CODE REVIEW

File: projects/observability/src/shared/components/interval-select/interval-select.component.ts [hunk 1]
```diff
@@ -66,27 +66,33 @@ export class IntervalSelectComponent implements OnChanges {
...
   }
 
-  private buildOptions(intervalValues: IntervalValue[] = []): IntervalOption[] {
-    return intervalValues.map(value => this.valueAsOption(value));
+  private buildOptions(values: IntervalValue[] = []): IntervalOption[] {
+    return values.map(value => this.valueAsOption(value, values));
   }
 
-  private valueAsOption(value: IntervalValue): IntervalOption {
+  private valueAsOption(value: IntervalValue, values: IntervalValue[]): IntervalOption {
```
Comment (anandtiwary) [inline projects/observability/src/shared/components/interval-select/interval-select.component.ts:73]: Could we renaming `values` to `availableIntervals`?

File: projects/observability/src/shared/dashboard/widgets/charts/cartesian-widget/cartesian-widget.model.ts [hunk 2]
```diff
@@ -112,6 +113,14 @@ export class CartesianWidgetModel<TData> {
   })
   public legendPosition: LegendPosition = LegendPosition.TopRight;
 
+  @ModelProperty({
+    key: 'max-data-points',
+    required: false,
+    displayName: 'Maximum Data Points',
+    type: NUMBER_PROPERTY.type
+  })
+  public maxDataPoints?: number;
```
Comment (anandtiwary) [inline projects/observability/src/shared/dashboard/widgets/charts/cartesian-widget/cartesian-widget.model.ts:122]: We are limiting at widget level. Does it mean 150 points per series? or overall?

Comment (aaron-steinfeld) [reply projects/observability/src/shared/dashboard/widgets/charts/cartesian-widget/cartesian-widget.model.ts:122]: I'd expect it to be per series, maybe a new name like maxSeriesDataPoints

Comment (anandtiwary) [reply projects/observability/src/shared/dashboard/widgets/charts/cartesian-widget/cartesian-widget.model.ts:122]: Yes.

**Pull request merged**